composer/trainer/trainer.py

```diff
@@ -1884,6 +1946,9 @@ def predict(self, dataloader: Union[DataLoader, DataSpec], subset_num_batches: i
                 self.state.outputs = self.state.model(self.state.batch)
                 self.engine.run_event(Event.PREDICT_AFTER_FORWARD)
 
+                if return_outputs:
+                    outputs.append(cpu_device.batch_to_device(self.state.outputs))
```

Comment:
should we concatenate along the batch dimension so its one tensor/np array?

Reply:
+1

Reply:
I was thinking about this.....and was leaning against it, as the outputs _could_ be different shapes. Or perhaps the output could be a dictionary. IDK, thoughts?

Reply:
Oh, good point... Don't we have smart infer for concat based on type, or is that just for split? I dont think this is a blocker for merging

Reply:
what about using a nestedtensor for outputs of different shapes? https://pytorch.org/docs/stable/nested.html

Reply:
I'd hesitate to have our public API depends on something in prototype stage at this point.

Reply:
I'll leave this as-is for now. We can always add in a collation argument later to try to consolidate it into one data structure. The user can pass a custom collate fn as part of their data spec, but we don't have a generic collate fn like we do with a batch split one.